drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -422,6 +374,76 @@ impl<S: SpiServer> ServerImpl<S> {
     }
 }
 
+impl<S: SpiServer> ServerImpl<S> {
+    fn do_send_recv_retries<'a>(
+        &mut self,
+        tx_size: usize,
+        timeout: u32,
+        retries: u16,
+    ) -> Result<Response, SprotError> {
+        let mut attempts_left = retries;
+
+        loop {
+            // Increase the error count here. We'll decrease if we return successfully.
+            self.io.stats.rx_invalid = self.io.stats.rx_invalid.wrapping_add(1);
```

Comment:
Uh, this should move into the match right before `Response::unpack`

Reply:
Better yet, we don't have to do this dance anymore. Just put it in the error branch.